Source/Extensions/Blazorise.DataGrid/_DataGridRow.razor.cs

```diff
@@ -88,14 +109,19 @@ protected internal Task OnCancelCommand()
             return Cancel.InvokeAsync( Item );
         }
 
+        protected internal Task OnMultiSelectCommand( bool IsSelected )
+        {
+            return MultiSelect.InvokeAsync( (IsSelected, Item) );
+        }
+
         #endregion
 
         #region Properties
 
         /// <summary>
         /// Indicates if the row is selected.
         /// </summary>
-        protected bool IsSelected => ParentDataGrid.EditState == DataGridEditState.None && (object)ParentDataGrid.SelectedRow == (object)Item;
+        protected bool IsSelected => ( ParentDataGrid.EditState == DataGridEditState.None && (object)ParentDataGrid.SelectedRow == (object)Item ) || ( ParentDataGrid.SelectedRows is object && ParentDataGrid.SelectedRows.Any( x => (object)( x ) == (object)Item ) );

```

Comment:
When you check for equality, use extension method `IsEqual`

Reply:
Agreed. Will make that change.
I implemented the equality as it already was implemented here: 
`(object)ParentDataGrid.SelectedRow == (object)Item;`